Data/ByteString.hs

```diff
@@ -1797,9 +1797,13 @@ useAsCString (BS fp l) action =
     pokeByteOff buf l (0::Word8)
     action (castPtr buf)
 
--- | /O(n) construction/ Use a @ByteString@ with a function requiring a @CStringLen@.
--- As for @useAsCString@ this function makes a copy of the original @ByteString@.
+-- | /O(n) construction/ Use a @ByteString@ with a function requiring a 'CStringLen'.
+-- As for 'useAsCString' this function makes a copy of the original @ByteString@.
 -- It must not be stored or used after the subcomputation finishes.
+--
+-- Beware that this function does not add a terminating @\NUL@ byte at the end of 'CStringLen'.
```

Comment:
It looks like the implementation actually does currently append a null byte, but I agree that doing so is not part of this function's contract.

```suggestion
-- Beware that this function is not required to add a terminating @\NUL@ byte at the end of the 'CStringLen' it provides.
```

Reply:
Right, it's `ShortByteString` version who does not add it, but `ByteString` one currently does.